src/HtsSystemContract.sol

```diff
@@ -38,6 +40,23 @@ contract HtsSystemContract is IERC20Events {
         assembly { accountId := sload(slot) }
     }
 
+    /**
+     * Query token info
+     * @param token - The token address to check
+     * @return responseCode - the response code for the status of the request. SUCCESS is `22`.
+     * @return tokenInfo - token info for `token`
+     */
+    function getTokenInfo(address token) htsCall external view returns (int64 responseCode, TokenInfo memory tokenInfo) {
+        require(token != address(0), "getTokenInfo: invalid token");
+
+        (bool success, bytes memory data) = token.staticcall(
```

Comment:
can we replace this `staticcall` with `IHederaTokenService(token).getTokenInfo(token)`? We would need to remove `view`, but this is how it's declared in `IHederaTokenService` so it should be fine.